test/bigml/test/api/source.clj

```diff
@@ -8,11 +8,13 @@
   (:use clojure.test))
 
 (defn- create-and-test [artifact & params]
-  (let [initial (api/get-final (apply source/create artifact params))
+  (let [{:keys [username api_key dev_mode] :as sticky-pars} params
```

Comment:
@sdesimone I think your understandable confusion is with the `:as` directive, which in fact assigns the entire expression being deconstructed, not just the ones listed for the keys. Put  `(let [{:keys [a b] :as p} {:a 1 :b 2 :c 3}] p)` in the repl and see.